[mlir] Update the return type of `getNum{Dynamic|Scalable}Dims`

banach-space opened the pull request:
Updates the return type of `getNumDynamicDims` and `getNumScalableDims`
from `int64_t` to `size_t`. This is for consistency with other
helpers/methods that return "size" and to reduce the number of
`static_cast`s in various places.

banach-space committed:
[mlir] Update the return type of `getNumDynamicDims`

Updates the return type of `getNumDynamicDims` from `int64_t` to
`size_t`. This is for consistency with other helpers/methods that return
"size" and to reduce the number of `static_cast`s in various places.

adam-smnk commented:
The `getNumDynamicDims` in `SparseTensorType` should also be updated for consistency.

adam-smnk commented:
And what about `getNumScalableDims` in vector type?

banach-space commented:
Thanks for taking a look!

> The getNumDynamicDims in SparseTensorType should also be updated for consistency.

> And what about getNumScalableDims in vector type?

Sending update shortly 👍🏻

banach-space committed:
fixup! [mlir] Update the return type of `getNumDynamicDims`

Update type.getNumDynamicDims in SparseTensor and getNumScalableDims in VectorType
